src/ddmd/func.d

```diff
@@ -1388,18 +1388,18 @@ extern (C++) class FuncDeclaration : Declaration
 
             // Precondition invariant
             Statement fpreinv = null;
-            if (addPreInvariant())
+            if (addPreInvariant() && isCtorDeclaration() is null)
```

Comment:
`addPreInvariant()` is a virtual function. Add a definition for it in `CtorDeclaration` for it instead that returns `false`.